build/yamls/base/conf/antrea-agent.conf

```diff
@@ -64,8 +64,18 @@ featureGates:
 # `antrea-agent` container must be set to the same value.
 #apiPort: 10350
 
-# Enable metrics exposure via Prometheus. Initializes Prometheus metrics listener.
-#enablePrometheusMetrics: false
+# EnablePrometheusMetrics is a map of metric categories to bool flags that enables or disables those metrics exposure.
+enablePrometheusMetrics:
+# Metrics in all below categories can be enabled or disabled through AllMetrics.
+#  AllMetrics: false
+# Metrics related to Pods. They can be enabled or disabled through PodMetrics.
+#  PodMetrics: false
+# Metrics related to Network Policies. They can be enabled or disabled through NetworkPolicyMetrics.
+#  NetworkPolicyMetrics: false
+# Metrics related to OVS switch. They can be enabled or disabled through OVSMetrics.
+#  OVSMetrics: false
+# Metrics related to connections when FlowExporter feature is enabled. They can be enabled or disabled through ConnectionMetrics.
+#  ConnectionMetrics: false
```

Comment:
what's the advantage of exposing all these configuration switches?

Reply:
unless this change is driven by some specific use case, I think we should look into addressing https://github.com/vmware-tanzu/antrea/issues/723 first.